src/main.ts

```diff
@@ -430,38 +442,38 @@ export default class ReadwisePlugin extends Plugin {
   ) {
     if (!this.settings.token) return;
 
-    const targetBookIds = bookIds || this.settings.booksToRefresh;
-
-    // add potentially-missing books to booksToRefresh (TODO - prob a lil inefficient? 🤷)
-    const knownFilesPaths = Object.keys(this.settings.booksIDsMap);
-    const shouldGetMissingBooks = this.settings.refreshBooks && !bookIds?.length;
-    if (shouldGetMissingBooks) {
-      for (const knownFilePath of knownFilesPaths) {
-        const file = this.app.vault.getAbstractFileByPath(knownFilePath);
-        if (!file) {
-          const bookId = this.settings.booksIDsMap[knownFilePath];
-          targetBookIds.push(bookId);
-        }
-      }
-    }
+    let targetBookIds = [
+      // try to sync provided bookIds
+      ...(bookIds || []),
 
-    const hasNeverSynced = !knownFilesPaths.length;
-    if (hasNeverSynced) {
-      this.notice("Preparing initial Readwise sync...", true);
-      await this.queueExport();
-      return;
+      // always try to sync failedBooks
+      ...this.settings.failedBooks,
+    ];
+
+    // only sync `booksToRefresh` items if "resync deleted files" enabled
+    if (this.settings.refreshBooks) {
+      targetBookIds = [
+        ...targetBookIds,
+        ...this.settings.booksToRefresh,
+      ];
     }
 
     if (!targetBookIds.length) {
-      console.log('Readwise Official plugin: no targetBookIds, triggering check for other updates...');
-      await this.queueExport(null, null, auto);
+      console.log('Readwise Official plugin: no targetBookIds, checking for new highlights');
+      // no need to hit refresh_book_export;
+      // just check if there's new highlights from the server

```

Comment:
nit: i'd leave a comment pointing out this is the "default" path if a user doesn't have any deleted files they need to resync